# Введение в Data Science и машинное обучение
# Задачи

## 1. Введение

### 1.4. Pandas, Dataframes, нет, панды тут ни при чем

А теперь используем эти методы на знаменитом [титаническом датасэте](https://stepik.org/media/attachments/course/4852/titanic.csv)! 

Загрузите датасэт, посмотрите на датафрэйм и ответьте на вопросы

Число колонок в представленном датафрэйме `_`, а число строк `_`. Тип float имеют `_` колонки, int - `_`, и object `_`

In [1]:
import pandas as pd

In [2]:
titanic = pd.read_csv('data\\titanic.csv')

In [3]:
titanic.shape

(891, 12)

In [4]:
titanic.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [5]:
titanic.dtypes.value_counts()

object     5
int64      5
float64    2
dtype: int64

Число колонок в представленном датафрэйме `12`, а число строк `891`. Тип float имеют `2` колонки, int - `5`, и object `5`

### 1.5. Фильтрация данных

`?`

У какой доли студентов из [датасэта](https://stepik.org/media/attachments/course/4852/StudentsPerformance.csv) в колонке `lunch` указано `free/reduced`?

In [6]:
students_performance = pd.read_csv('data\\StudentsPerformance.csv')

In [7]:
students_performance.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [8]:
students_performance.columns

Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='object')

In [9]:
len(students_performance[students_performance['lunch'] == 'free/reduced']) \
    / len(students_performance)

0.355

`?`

Как различается среднее и дисперсия оценок по предметам у групп студентов со стандартным или урезанным ланчем?

In [10]:
students_performance[students_performance['lunch'] == 'free/reduced'].describe()

math score  reading score  writing score
count  355.000000     355.000000     355.000000
mean    58.921127      64.653521      63.022535
std     15.159956      14.895339      15.433823
min      0.000000      17.000000      10.000000
25%     49.000000      56.000000      53.000000
50%     60.000000      65.000000      64.000000
75%     69.000000      75.000000      74.000000
max    100.000000     100.000000     100.000000

In [11]:
students_performance[students_performance['lunch'] != 'free/reduced'].describe()

math score  reading score  writing score
count  645.000000     645.000000     645.000000
mean    70.034109      71.654264      70.823256
std     13.653501      13.830602      14.339487
min     19.000000      26.000000      22.000000
25%     61.000000      63.000000      62.000000
50%     69.000000      72.000000      72.000000
75%     80.000000      82.000000      81.000000
max    100.000000     100.000000     100.000000

In [12]:
students_performance[students_performance['lunch'] == 'free/reduced'].var()

math score       229.824270
reading score    221.871139
writing score    238.202881
dtype: float64

In [13]:
students_performance[students_performance['lunch'] != 'free/reduced'].var()

math score       186.418089
reading score    191.285560
writing score    205.620887
dtype: float64

- `студенты со стандартным ланчем имеют в среднем более хорошие оценки`

- `дисперсия оценок студентов со стандартным ланчем меньше`